### DISCUSSION TITLE
Iframing OTT to my website no longer works

### DISCUSSION THREAD

**Discussion by maddie-wang** [dyc3/opentogethertube#402] [General]
Hi OTT!

I've been using OTT (iframe) with my video calling project Sesame (https://sesamecall.com) for a few days now. Me and many other students love using it to listen to music or doing workouts together in a call. We have about 400 calls per day!

I noticed today that iframing OTT no longer works. (If you go to https://sesamecall.com/call/?room=test-call-123 and click the youtube button in the upper left you'll see what I mean)

Error: `Refused to display 'https://opentogethertube.com/' in a frame because it set 'X-Frame-Options' to 'sameorigin'`

I was wondering whether OTT would be open to enable iframe embedding to their site, so that more people can enjoy  music/videos together while in a call

Also I recognize my site might be sending too many requests to create permanent rooms to OTT, since I noticed a few heroku errors pop up. If that's the case, I'm sorry about any down time or problems I've caused. It's my fault and I know it's frustrating when another player messes up with your system, and no one deserves that. I'm very willing to hear more about the issues there and fully mitigate the issues immediately, so it doesn't happen again. Also super down to partner up, if that's a thing too 😀 

@dyc3

**Comment by dyc3** [comment]
That would explain why my traffic shot up and hasn't gone down since.

> Also I recognize my site might be sending too many requests to create permanent rooms to OTT.

I would very much prefer if you created temporary rooms instead. Creating permanent rooms hits the database, and temporary rooms do not. I have a limited number of rows I can use. Also, if you could add a `description` to the rooms you create to indicate they're from Sesame, that would be great.

I didn't _intentionally_ disable iframes, I've had cloudflare on "under attack" mode while I added a bit more optimization and logging. I'm assuming that sets the `X-Frame-Options` header. It's disabled now.

Having OTT in an iframe does sound a little janky, and it'll be subject to any layout changes that I make in the future. I'd be open to any PRs that might make your use case easier or less janky.

**Reply by maddie-wang** [reply]
Thanks so much for responding so quickly omg 🍡 😀 

Awesome, I just finished rewriting my code to create temporary rooms 👍  Changed temporary from false to true.

```
      axios
        .post(`https://opentogethertube.com/api/room/create`, {
        description: "",
        name: ottID,
        queueMode: "manual",
        temporary: true,
        title: "",
        visibility: "unlisted",
      };)

```

YAY looks like the iframes work again :D

Let me know how else I can help!  Thanks so much @dyc3  you are so amazing! Yeah I agree the iframe is pretty janky, and regardless, I appreciate how open you are to our jank usecase hehe :) Have an amazing day!!!

**Reply by maddie-wang** [reply]
PS  - we're listening to OTT (Lofi music) right now and coworking !! :O :D 

if you're free in the next hour or two, and wanna say hi, we're in here https://www.sesamecall.com/call/?room=814828dd-4102-4470-85ed-c88cd634049e

**Reply by dyc3** [reply]
Thanks @maddie-wang, is there a better place I can contact you in the future?

**Reply by dyc3** [reply]
I would prefer an email, twitter, or discord, but this works too. Actually, edit history is preserved on github, so better to delete the message entirely.
